include/deal.II/numerics/data_postprocessor.h

```diff
@@ -422,8 +450,47 @@ namespace DataPostprocessorInputs
  * class.
  *
  *
+ * <h3>Complex-valued solutions</h3>
+ *
+ * There are PDEs whose solutions are complex-valued. For example, step-62
+ * solves a problem whose solution at each point consists of a complex number
+ * represented by a `std::complex<double>` variable. (step-29 also solves such
+ * a problem, but there we choose to represent the solution by two real-valued
+ * fields.) In such cases, the vector that is handed to
+ * DataOut::build_patches() is of type `Vector<std::complex<double>>`, or
+ * something essentially equivalent to this. The issue with this, as also
+ * discussed in the documentation of DataOut itself, is that no visualization
+ * program we know of can actually represent complex quantities. The only thing
+ * that can reasonably be stored in data files for visualization tools are
+ * real-valued quantities.
```

Comment:
I don't think this is completely accurate. The software based on the VTK library can not deal with complex numbers, but other visualization software, such as `python/numpy` have native support for complex numbers. HDF5 can store `std::complex` datasets. The `deal.II` interface of HDF5 supports `std::complex` datasets.

In step-62 I store complex-valued data in a HDF5 file and then visualize the data with `python`.

I would say that the VTK library does not support complex numbers, but I wouldn't say that the rest of the visualization tools don't support complex numbers.

Reply:
I agree that other file formats allow *storing* complex numbers, but I don't know any visualization file format that allows *visualizing* that data.

Certainly, to the best of my knowledge, none of the file formats that the functions in `DataOutBase` can write in, supports complex numbers.

I'm willing to make this sound more conciliatory, of course, but I think it's fundamentally true: All visualization software ultimately visualizes real-valued data :-)

Reply:
> I'm willing to make this sound more conciliatory.

That sounds good :)